hexkit/protocols/eventpub.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2021 - 2022 Universität Tübingen, DKFZ and EMBL
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Protocols related to event handling."""
+
+from typing import Protocol
+
+from hexkit.custom_types import JsonObject
+
+
+class EventPublisherProtocol(Protocol):
```

Comment:
The ASCII check from the akafka provider and the NonAsciiStrError could be moved here if we decide to use an ABC instead.

Reply:
Yes, exactly. Could we maybe implement that in another PR? I already made a ticket for that: [GDEV-775](https://jira.verbis.dkfz.de/browse/GDEV-775)